lib/mox/server.ex

```diff
@@ -0,0 +1,71 @@
+defmodule Mox.Server do
+  @moduledoc false
+
+  use GenServer
+
+  # Public API
+
+  def start_link(_options) do
+    GenServer.start_link(__MODULE__, %{}, name: __MODULE__)
+  end
+
+  def add_expectation(key, {n, calls}) do
+    GenServer.call(__MODULE__, {:add_expectation, key, {n, calls}})
+  end
+
+  def fetch_expectation_for_dispatch(key) do
```

Comment:
This name is a bit incorrect because we are not only fetching but we are also decreasing the counter.

Reply:
I will simplify this by using the same function head but with extra parameter stating clearly if the counter should be updated or not.